app/views/site/_view_gene_expression_heatmap.html.erb

```diff
@@ -152,18 +152,14 @@
             url: "<%= get_new_annotations_path(study: params[:study])%>?cluster=" + newCluster,
             method: 'GET',
             dataType: 'script',
-            success: function (data) {
-                // parse response as a string and see if currently selected annotation exists in new annotations
-                if (data.indexOf(currAnnot) >= 0) {
-                    $('#annotation').val(currAnnot);
-                }
-                $(document).ready(function () {
+            complete: function (jqXHR, textStatus) {
+                if (textStatus === 'success') {
                     // since we now have new annotations, we need to set them in the search form for persistence
                     var an = $('#annotation').val();
```

Comment:
seems like these 3 lines occur in 3 different places.  Is there a sensible place to pull these out into an 'updateAnnotationValues' method or somesuch?

Reply:
I had wanted to do that, but there was enough difference between all the various instances of this callback that it seemed like abstraction would be messy.  But I'll take another look at it to see if there's a clean way of doing it.